Tests/IntegrationTests/Roslyn/RoslynGenTests.cs

```diff
@@ -92,6 +92,92 @@ public void RoslynPlugin_GenerateForValidAnalyzer_Succeeds()
         }
 
+        // Test verifies https://jira.sonarsource.com/browse/SFSRAP-32
+        [TestMethod]
+        public void RoslynPlugin_GenerateForDependencyAnalyzers_Succeeds()
+        {
+            // Arrange
+            TestLogger logger = new TestLogger();
+            string outputDir = TestUtils.CreateTestDirectory(this.TestContext, ".out");
+            string dummyContentFile = TestUtils.CreateTextFile("dummy.txt", outputDir, "non-analyzer content file");
+
+            // Create a valid analyzer package
+            RoslynAnalyzer11.CSharpAnalyzer analyzer = new RoslynAnalyzer11.CSharpAnalyzer();
+
+            string fakeRemoteNuGetDir = TestUtils.CreateTestDirectory(this.TestContext, ".fakeRemoteNuGet");
+            IPackageManager fakeRemotePkgMgr = CreatePackageManager(fakeRemoteNuGetDir);
+            IPackage child1 = AddPackage(fakeRemotePkgMgr, "Analyzer.Child1", "1.1.0", analyzer.GetType().Assembly.Location);
+            IPackage child2 = AddPackage(fakeRemotePkgMgr, "Analyzer.Child2", "1.2.0", analyzer.GetType().Assembly.Location);
+            IPackage targetPkg = AddPackage(fakeRemotePkgMgr, "Empty.Parent", "1.0.0", dummyContentFile, child1, child2);
+
+            string localPackageDestination = TestUtils.CreateTestDirectory(this.TestContext, ".localpackages");
+
+            // Act
+            NuGetPackageHandler nuGetHandler = new NuGetPackageHandler(fakeRemotePkgMgr.LocalRepository, localPackageDestination, logger);
+            AnalyzerPluginGenerator apg = new AnalyzerPluginGenerator(nuGetHandler, logger);
+            ProcessedArgs args = new ProcessedArgs(targetPkg.Id, targetPkg.Version, "cs", null, false, 
+                true /* generate plugins for dependencies with analyzers*/, outputDir);
+            bool result = apg.Generate(args);
+
+            // Assert
+            Assert.IsTrue(result);
+            string[] jarFilePaths = AssertPluginJarsExist(outputDir, 2); // Expecting one plugin per dependency with analyzers
+
+            foreach (string jarFilePath in jarFilePaths)
+            {
+                // Check the content of the files embedded in the jar
+                ZipFileChecker jarChecker = new ZipFileChecker(this.TestContext, jarFilePath);
+
+                // Check the contents of the embedded config file
+                string embeddedConfigFile = jarChecker.AssertFileExists("org\\sonar\\plugins\\roslynsdk\\configuration.xml");
+                RoslynSdkConfiguration config = RoslynSdkConfiguration.Load(embeddedConfigFile);
+
+                // Check the config settings, find which of the dependencies the jar is for (so that we can check the correct strings)
+                IPackage originalPkg = null;
+                if (config.PluginKeyDifferentiator.Equals("analyzerchild1"))
+                {
+                    originalPkg = child1;
+                }
+                if (config.PluginKeyDifferentiator.Equals("analyzerchild2"))
```

Comment:
Trying to work out what the current instance is inside a loop like this is a code smell. The checking code should be factored out and parameterised e.g. by passing in the package. It looks like all of this code was copied from the existing test, so that test should be refactored too.